backend/api/routers/session.py

```diff
@@ -0,0 +1,57 @@
+from fastapi import APIRouter, HTTPException, Depends
+from sqlalchemy.orm import Session
+
+import uuid
+
+from rope.api.auth import verify_google_token, verify_user
+from rope.api import database
+from rope.api.sessions import (
+    create_session,
+    destroy_session,
+    get_request_session,
+    get_session,
+)
+from rope.api.models import GoogleLoginData
+
+router = APIRouter(
+    prefix="/session",
+    tags=["session"],
+)
+
+
+@router.post("/")
```

Comment:
Actually, I realized something pretty major in terms of regression :sweat_smile: : I think these routers broke the contract expected by the FE. Perhaps it has to do with the trailing `/`.

Reply:
Let me mess around with the trailing `/` and see if the login flow comes back. If it's that simple to fix on the BE, then probably makes sense to provide the fix there as opposed to updating the FE code (unless something about routers inherently requires FE updates).

We'll have to move those `fetch()` calls into `ropeApi.ts` maybe in the next FE issue or in a separate issue. Probably easy enough to just combine into a currently written issue.

Reply:
I agree with trying to see if / how we can preserve existing behavior w/o changing FE code.

I think I will just spin up a dedicated issue for the minor refactoring to consolidate all API calls into `ropeApi.ts`. It should be a quick `1`, but that way it's not adding scope to existing issues.

Reply:
The following actions were done inside `session.py`:

I've tried removing the trailing `/`, running `docker compose down` and `docker compose up --build -d`, but the login flow remained broken. I also removed the `prefix` set via `APIRouter()` and set the path at the endpoint level. I ran the same docker commands after applying the second fix, but that did not work either.

Reply:
I think I found a potential way to re-align: The root issue is probably that we did two things when creating routers:

* We seem to have added a trailing '/` which matters
* We are setting a prefix on routers which need to be able to specify routes without a trailing `/`

A way to fix this would be remove the `prefix` argument for routers (except for perhaps `admin.py` since we always append something to that prefix) + make sure all routes don't end in `/` as they were before.

Reply:
Let's also update our `test_client` fixture so these regressions get captured:

```python
@pytest.fixture
def test_client():
    return TestClient(app, follow_redirects=False)
```

Reply:
We opted with removing the prefix, as there are some instances where if there is nothing to append to the prefix then you would end up with something like `router.get("")`. It was not obvious that the prefix was being insert into the empty quotes. For improved readability + consistency, we removed the prefixes.